api/auth.go

```diff
@@ -78,14 +117,51 @@ func authenticationHandler(w http.ResponseWriter, r *http.Request) bool {
 			w.WriteHeader(http.StatusUnauthorized)
 			return false
 		}
-	}
 
-	user, err := helpers.Store(r).GetUser(userID)
-	if err != nil {
-		if err != db.ErrNotFound {
-			// internal error
-			log.Error(err)
+		user, err = fetchUser(helpers.Store(r), userID)
+		if err != nil {
+			w.WriteHeader(http.StatusUnauthorized)
+			return false
 		}
+
+		break
+	case authExternal:
+		username := strings.ToLower(r.Header.Get(util.Config.ExternalAuthHeader.Username))
+
+		// find user account
+		user, err := helpers.Store(r).GetUserByLoginOrEmail(username, username)
+		if err != nil {
+			if err != db.ErrNotFound {
+				log.Error(err)
+				w.WriteHeader(http.StatusUnauthorized)
+				return false
+			}
+
+			log.Debug(username + " does not exist yet, creating it")
+
+			externalAuthUser := db.UserWithPwd{
+				User: db.User{
+					Username: username,
+					Created:  time.Now(),
+					Name:     username,
+					Email:    username + "@example.org", // FIXME
```

Comment:
Not entirely sure how to deal with that yet. Maybe have to read how LDAP _works_. But I guess there's also generally more data available from an external directory.